include/swift/AST/ASTMangler.h

```diff
@@ -83,6 +83,14 @@ class ASTMangler : public Mangler {
   /// a critical role.
   bool AllowIsolatedAny = true;
 
+  /// If enabled, typed throws can be encoded in the mangled name.
+  /// Suppressing type attributes this way is generally questionable ---
+  /// for example, it does not interact properly with substitutions ---
+  /// and should only be done in situations where it is just going to be
+  /// interpreted as a type and the exact string value does not play
+  /// a critical role.
+  bool AllowTypedThrows = true;
```

Comment:
Perhaps we should sink the RuntimeCompatVersion into the ASTMangler eventually

Reply:
Yeah. I didn't want to rock the boat for 6.0, but if we can do that and then have some way of ensuring that new type metadata features opt in there, it would be much better for future evolution.